File: Operacje na tekście.py
text = "Ala ma kota, a kot ma Ale."

print(f"Oryginalny tekst: {text}")

# W Pythonie napisy są niezmienne, więc metody tworzą nową wersję tekstu
text_lower = text.lower()
print(f"Małe litery:       {text_lower}")

text_upper = text.upper()
print(f"Wielkie litery:    {text_upper}")

print(f"Długość tekstu:    {len(text)} znaków")

File: Operacje na tekście.c
#include <stdio.h>

void toLower(char text[]) {
    int i = 0;
    while (text[i]) {
        if ('A' <= text[i] && text[i] <= 'Z') {
            text[i] += ('a' - 'A');
        }
        i++;
    }
}

void toUpper(char text[]) {
    int i = 0;
    while (text[i]) {
        if ('a' <= text[i] && text[i] <= 'z') {
            text[i] -= ('a' - 'A');
        }
        i++;
    }
}

int textSize(char text[]) {
    int size = 0;
    while (text[size] != '\0') {
        size++;
    }
    return size;
}

int main() {
    char text[] = "Ala ma kota, a kot ma Ale.";
    
    printf("Oryginalny tekst: %s\n", text);
    
    toLower(text);
    printf("Male litery:       %s\n", text);
    
    toUpper(text);
    printf("Wielkie litery:    %s\n", text);
    
    printf("Dlugosc tekstu:    %d znakow\n", textSize(text));

    return 0;
}
